event_track_location_overlap/models/event_track_location.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2017 Tecnativa - Jairo Llopis
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from datetime import timedelta
+from odoo import _, api, exceptions, fields, models
+
+
+class EventTrackLocation(models.Model):
+    _inherit = "event.track.location"
+
+    overlappable = fields.Boolean(
+        help="Can this location have simultaneous tracks?"
+    )
+
+    # TODO oca.decorators.foreach() in v12
+    @api.one  # pragma pylint: disable=api-one-deprecated
+    @api.constrains("overlappable")
+    def _check_overlappable(self):
+        """Ensure no overlaps happen with this location."""
+        # Skip locations that can be overlapped
+        if self.overlappable:
+            return
+        # Get track states that can always overlap
+        skip_track_states = self.env.context.get(
+            "skip_track_states",
+            ["draft", "refused", "cancel"],
+        )
+        # Get tracks that could produce an overlap
+        remaining_tracks = self.env["event.track"].search([
+            ("location_id", "=", self.id),
+            ("state", "not in", skip_track_states),
```

Comment:
This field is no more in `event.track` model. Put `("event_id.state", "not in", ...`